code/modules/vehicles/cars/vim.dm

```diff
@@ -0,0 +1,92 @@
+ */
+/obj/vehicle/sealed/car/vim
+	name = "\improper Vim"
+	desc = "An minature exosuit from Nanotrasen, developed to let the irreplacable station pets live a little longer."
+	icon_state = "vim"
+	max_integrity = 50
+	armor = list(MELEE = 70, BULLET = 40, LASER = 40, ENERGY = 0, BOMB = 30, BIO = 0, RAD = 0, FIRE = 80, ACID = 80)
+	enter_delay = 20
+	movedelay = 0.6
+	engine_sound_length = 0.3 SECONDS
+	light_system = MOVABLE_LIGHT_DIRECTIONAL
+	light_range = 4
+	light_power = 2
+	light_on = FALSE
+	engine_sound = 'sound/effects/servostep.ogg'
+	///TRUE while the vim is being welded
+	var/being_repaired = FALSE
+	COOLDOWN_DECLARE(sound_cooldown)
+
+/obj/vehicle/sealed/car/vim/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] can be repaired with a welder.")
+
+/obj/vehicle/sealed/car/vim/obj_destruction(damage_flag)
+	new /obj/effect/decal/cleanable/oil(get_turf(src))
+	do_sparks(5, TRUE, src)
+	visible_message(span_boldannounce("[src] blows apart!"))
+	return ..()
+
+/obj/vehicle/sealed/car/vim/mob_try_enter(mob/entering)
+	if(!isanimal(entering))
+		return FALSE
+	var/mob/living/simple_animal/animal = entering
+	if(animal.mob_size != MOB_SIZE_TINY)
+		return FALSE
+	return ..()
+
+/obj/vehicle/sealed/car/vim/welder_act(mob/living/user, obj/item/tool)
+	. = ..()
+	. = TRUE
+	if(!tool.tool_start_check(user))
+		return
+	if(being_repaired)
+		user.balloon_alert(user, "already being repaired!")
+		return
+	if(obj_integrity == max_integrity)
+		user.balloon_alert(user, "already fully repaired!")
+		return
+
+	user.balloon_alert(user, "repairing [src]...")
+	audible_message(span_hear("You hear welding."))
+	being_repaired = TRUE
+	if(!tool.use_tool(src, user, 3 SECONDS, volume=50))
+		being_repaired = FALSE
+		user.balloon_alert(user, "interrupted!")
+		return
+	being_repaired = FALSE
+
+	obj_integrity = min(obj_integrity + VIM_HEAL_AMOUNT, max_integrity)
+	user.balloon_alert(user, "[obj_integrity == max_integrity ? "fully " : ""]repaired [src]")
+
+/obj/vehicle/sealed/car/vim/mob_enter(mob/newoccupant, silent = FALSE)
+	. = ..()
+	update_appearance()
+	playsound(src, 'sound/machines/windowdoor.ogg', 50, TRUE)
+	if(obj_integrity == max_integrity)
+		SEND_SOUND(newoccupant, sound('sound/mecha/nominal.ogg',volume=50))
+
+/obj/vehicle/sealed/car/vim/generate_actions()
+	initialize_controller_action_type(/datum/action/vehicle/sealed/climb_out/vim, VEHICLE_CONTROL_DRIVE)
+	initialize_controller_action_type(/datum/action/vehicle/sealed/noise/chime, VEHICLE_CONTROL_DRIVE)
+	initialize_controller_action_type(/datum/action/vehicle/sealed/noise/buzz, VEHICLE_CONTROL_DRIVE)
+	initialize_controller_action_type(/datum/action/vehicle/sealed/headlights/vim, VEHICLE_CONTROL_DRIVE)
+
+/obj/vehicle/sealed/car/vim/update_overlays()
+	. = ..()
+	var/static/piloted_overlay
+	var/static/headlights_overlay
+	if(isnull(piloted_overlay))
+		piloted_overlay = iconstate2appearance(icon, "vim_piloted")
+		headlights_overlay = iconstate2appearance(icon, "vim_headlights")
```

Comment:
```suggestion
	var/static/piloted_overlay = iconstate2appearance(icon, "vim_piloted")
	var/static/headlights_overlay = iconstate2appearance(icon, "vim_headlights")
```
Static values in proc calls will be initialized a single time.

Reply:
Bad suggestion.

Static values in proc calls are initialized at world load time with a src of null.

Reply:
![image](https://user-images.githubusercontent.com/7069733/126582749-7ee318ca-ffe8-4b44-9963-b5bff8f968e9.png)

Reply:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#icons-are-for-image-manipulation-and-defining-an-objs-icon-var-appearances-are-for-everything-else

Reply:
I was not aware of this, very nice.